src/expr/impl/src/udf/external.rs

```diff
@@ -174,9 +175,16 @@ fn get_or_create_flight_client(link: &str) -> Result<Arc<Client>> {
         // reuse existing client
         Ok(client)
     } else {
+        static RUNTIME: LazyLock<Runtime> = LazyLock::new(|| {
+            tokio::runtime::Builder::new_multi_thread()
+                .thread_name("rw-udf")
+                .enable_all()
+                .build()
+                .expect("failed to build udf runtime")
+        });
         // create new client
         let client = Arc::new(tokio::task::block_in_place(|| {
-            tokio::runtime::Handle::current().block_on(async {
+            RUNTIME.block_on(async {
```

Comment:
The current thread will still be blocked until the client is created. I don't think this really helps. 🤔

Reply:
I am also curious why it will block the whole runtime. I have met the same issue when developing `file_scan`, but when I used a dedicated runtime, the hung problem never happened. I'm not quite sure of the root cause right now.

Reply:
Can we simply make this function `async`? 🥵

Reply:
It seems impossible to make this function `async` 🥵

Reply:
I think we can follow a similar strategy in meta. That is, start a new service in frontend, which will manage async connection with udf-server / iceberg / pg.

Then, inside the binder we just use `UnboundedSender` to send requests over to this service, which can be done in a synchronous way, since the send buffer is unbounded.

Reply:
Reason I'm suggesting this is because I encountered the same issue here: https://github.com/risingwavelabs/risingwave/pull/18811/files#r1792074116. But seems like in the case of pg it just gets totally stuck, even the runtime hack doesn't work.

Reply:
I guess we can support it after we support `create connection`.